backend/app/views/spree/admin/menus/edit.html.erb

```diff
@@ -45,7 +45,7 @@
                 <% end %>
               <% else %>
                 <div class="alert alert-info no-objects-found m-5">
-                  <%= raw Spree.t('admin.navigation.no_menu_items', menu: @menu.name) %>
+                  <%= raw Spree.t('admin.navigation.no_menu_items', menu: @menu.root.name) %>
```

Comment:
```suggestion
                  <%= raw Spree.t('admin.navigation.no_menu_items', menu: @menu.root_name) %>
```

maybe let's delegate this to root or create a method to check if root exists and if not fallback to menu.name?

Reply:
One moment...

Reply:
Hi @damianlegawiec ,

There should never be a menu without a menu having a root, it's essential that a menu has a root, and quite essential that the root name is the same as the menu name, this is why I changed this so that a test chan check that on the view side.


**Explanation** 
The root of the menu is created on menu create, and its name kept in sync.
```ruby
    after_create :set_root
    after_save :update_root_name

    def set_root
      self.root ||= MenuItem.create!(menu_id: id, name: name, item_type: 'Container')
    end

    def update_root_name
      return unless saved_change_to_name?

      root.update(name: name)
    end
```
There is no way in the admin UI to destroy the menu root, only when a menu is destroyed and its dependents are also destroyed.

There is good reason that the root requirement for organising the items, but also, the reason for the name sync is that here:
```ruby
    MENU_LOCATIONS_PARAMETERIZED.each do |name|
      define_singleton_method("for_#{name}") do |locale|
        menu = find_by(location: name, locale: locale.to_s) || find_by(location: name)
        if menu.present?
          menu.root
        end
      end
    end
```

We are returning the ```menu.root```, to disguise the roots existence, therefor any unsuspecting front end devwho is familia with other platforms can call ```<%= spree_menu.name %>``` and to get straight to the menu items they would expect, without having to go through the root itself.

But the dev can still call ```<%= spree_menu.name %>``` and it returns the menu name, when in reality they are getting the ```menu.root.name``` that is synced with the ```menu.name```.

Reply:
In light of this, I can change it back to `@menu.name` or leave it as `@menu.root.name` and have that little test in there to make sure that the menu name is in sync.

But I think having a fallback, is not really required as there should never be a menu without a root.